lib/versions_repository.py

```diff
@@ -138,3 +139,118 @@ def read_version_and_milestone(versions_repo):
+
+                LOG.debug("Updating package {name} to {evra}".format(
+                    name=package.name, evra=package_evra))
+                spec_file_contents += SPEC_DEPENDENCY_TEMPLATE.format(
+                    name=package.name, evra=package_evra)
+
+    if spec_updated:
+        with open(spec_file_path, "w") as spec_file:
+            spec_file.write(spec_file_contents)
+
+
+def update_metapackage(
+        versions_repo, distro, metapackage_name, packages_names,
+        user_name, user_email):
+    """
+    Update package version and dependencies to the packages created by
+    a build of the specified versions repository commit.
+
+    Args:
+        versions_repo (GitRepository): versions Git repository handler
+        distro (distro.LinuxDistribution): Linux distribution
+        metapackage_name (str): name of the release package whose
+            dependencies will be updated
+        packages_names ([str]): list of dependencies of the release
+            package
+        user_name (str): name of the user updating the spec file
+        user_name (str): email of the user updating the spec file
+    """
+    LOG.info("Updating release package dependencies: "
+             "{}".format(", ".join(packages_names)))
+    pm = packages_manager.PackagesManager(packages_names)
+    # TODO: this is coupled with RPM-based Linux distributions
+    pm.prepare_packages(packages_class=rpm_package.RPM_Package,
+                        download_source_code=False, distro=distro)
+
+    LOG.info("Updating release package YAML file")
+    YAML_START_DELIMITER = "    install_dependencies:"
+    metapackage_yaml_file_path = os.path.join(
+        versions_repo.working_tree_dir, metapackage_name,
+        metapackage_name + ".yaml")
+    yaml_install_dependencies_string = "{start}\n{contents}".format(
+        start=YAML_START_DELIMITER,
+        contents=create_yaml_install_dependencies_string(pm.packages))
+    replace_file_section(
+        metapackage_yaml_file_path, yaml_install_dependencies_string,
+        YAML_START_DELIMITER)
+
+    LOG.info("Updating release package spec file")
+    version_file_path = os.path.join(versions_repo.working_tree_dir, "VERSION")
+    with open(version_file_path) as version_file:
+        version_parts = version_file.readlines()[-1].strip().split("-")
+    metapackage_version = version_parts.pop(0)
+    if version_parts:
+        metapackage_milestone = version_parts.pop(0)
+    else:
+        metapackage_milestone = "%nil"
+
+    metapackage = rpm_package.RPM_Package.get_instance(
+        metapackage_name, distro)
+    metapackage.spec_file.update_version(metapackage_version)
+    metapackage.spec_file.replace_macro_definition(
+        "milestone", metapackage_milestone)
+
+    replace_spec_dependencies(metapackage.spec_file.path)
+    metapackage.spec_file.bump_release(
+        ["Update package dependencies"], user_name, user_email)
```

Comment:
IMHO these metapackage rpm spec handling methods should not be in versions_repository.py. As you want to reuse them, I suggest you create a new Python module.